org.lflang/src/org/lflang/generator/JavaGeneratorUtils.java

```diff
@@ -222,8 +227,27 @@ public static void validateImports(
             if (
                 bad.contains(resource) || issues.size() > 0
             ) {
+                // Report the error on this resource.
+                // Need a path, which, sadly, seems impossibly difficult to get reliably.
+                // Be very paranoid here because we don't want errors to occur while error reporting.
+                // FIXME: Is this how to get a path?
+                Path path = Paths.get("Unknown file");
+                if (resource.getURI().hasPath()) {
+                    String filename = resource.getURI().path();
+                    if (filename != null) {
+                        path = Paths.get(filename);
+                    }
+                }
+                for (Issue issue : issues) {
+                    errorReporter.reportError(path, issue.getLineNumber(), issue.getMessage());
+                }
+                
+                // Report errors on resources that import this one.
                 for (Reactor downstreamReactor : instantiationGraph.getDownstreamAdjacentNodes(reactor)) {
                     for (Import importStatement : ((Model) downstreamReactor.eContainer()).getImports()) {
+                        // FIXME: This will report the error on ALL import statements in
+                        // file doing the importing, not just the one importing this resource.
+                        // I have no idea how to determine which import statement is the right one.
```

Comment:
The classes on the left-hand side of the `from` of an import statement should be linked to a class definition in some resource by our scope provider. If the resource matches the resource currently being validated, then that statement should be marked as erroneous.

Reply:
I don't understand this suggestion. The resource will not match the resource currently being validated because it is an imported file. In any case, this bug is not new to this PR, but was in the code before. I just added the FIXME. I'll open an issue for this and move on.

Reply:
OK, that works too.